pusher/crypto.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import (
+    print_function,
+    unicode_literals,
+    absolute_import,
+    division)
+
+import hashlib
+import nacl
+import base64
+
+from pusher.util import (
+    ensure_text,
+    ensure_binary,
+    data_to_string)
+
+import nacl.secret
+import nacl.utils
+
+# The prefix any e2e channel must have
+ENCRYPTED_PREFIX = 'private-encrypted-'
+
+def is_encrypted_channel(channel):
+    """
+    is_encrypted_channel checks() if the channel is encrypted by verifying the prefix
+    """
+    if channel.startswith(ENCRYPTED_PREFIX):
+        return True
+    return False
+
+def is_encryption_master_key_valid(encryption_master_key):
+    """
+    is_encryption_master_key_valid() checks if the provided encryption_master_key is validby checking its length
+    """
+    if len(encryption_master_key) == 32:
+        return True
```

Comment:
I think that we should check whether the key satisfies the criteria of `ensure_binary` before checking the length. Otherwise, we could get a Unicode key that expands into more bytes than expected.
```py
( len(u'a_key') == len(u'مفتاح') ) == True  # 5 letters each
len(u'a_key'.encode('utf-8')) == 5  # 5 bytes
len(u'مفتاح'.encode('utf-8')) == 10  # 10 bytes
```

Reply:
Yes, we are already checking it outside `is_encryption_master_key_valid()`, we wouldn't want to check it at this level since this is outside the scope of this function.

Reply:
In this case, how about marking the function as private?